core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -72,9 +72,47 @@ private[spark] case class ExecutorDeadException(message: String)
 /**
  * Exception thrown when Spark returns different result after upgrading to a new version.
  */
-private[spark] class SparkUpgradeException(version: String, message: String, cause: Throwable)
+private[spark] class SparkUpgradeException(
+    version: String,
+    message: String,
+    cause: Throwable,
+    errorClass: Option[String],
+    messageParameters: Array[String])
   extends RuntimeException("You may get a different result due to the upgrading of Spark" +
-    s" $version: $message", cause)
+    s" $version: $message", cause) with SparkThrowable {
+
+  def this(version: String, message: String, cause: Throwable) =
+    this(
+      version = version,
+      message = message,
+      cause = cause,
+      errorClass = None,
+      messageParameters = Array.empty
+    )
+
+  def this(errorClass: String, messageParameters: Array[String], cause: Throwable) =
```

Comment:
I don't know if we want to have this constructor - it feels like the version should always be provided.